js/plates-model/earthquake.js

```diff
@@ -33,13 +34,16 @@ export default class Earthquake {
       // "Deep" earthquakes take place around lithosphere of the subducting plate, so where the collision happens.
       // Earthquakes are always attached to the top plate, as that's where they are felt and registered. That makes
       // visualization better too, as earthquakes won't be moving together with the subducting plate.
-      const deep = subductingField && random() < 0.5
-      const baseField = deep ? subductingField : field
-      const availableRange = baseField.crustThickness + baseField.lithosphereThickness - MIN_DEPTH
-      const earthquakeElevation = baseField.elevation - MIN_DEPTH - random() * availableRange
-      this.depth = field.elevation - earthquakeElevation
+      // Also, make sure that shallow earthquakes don't show up in trench-areas, as cross section would look confusing.
+      const shallow = (field.elevation - field.crustThickness) > subductingField.elevation & random() < SHALLOW_EQ_PROBABILITY
+      const baseField = shallow ? field : subductingField
+      const availableRange = baseField.crustThickness + (shallow ? 0 : baseField.lithosphereThickness) - MIN_DEPTH
+      this.depth = Math.max(
```

Comment:
I'm noticing that some earthquakes in cross-section appear to be almost entirely outside of the lithosphere region, with only a very small overlap. I'm not sure if we need to account for earthquake size in the calculations to account for this so that the center point of an earthquake remains always within that zone - Amy may have thoughts on this.

Reply:
Sometimes it happens because we smooth and average field elevation using a few fields around. If you move cross-section line slightly up or down, the relative position of earthquake and lithosphere might change. It'll be really difficult to improve, so I'd wait for Amy's feedback - if it's a problem or not. Personally, I don't think it's worth the effort, as these differences are small, and on some pictures in the internet earthquakes are pretty spread out.

Reply:
I think this issue has been mostly fixed. Amy noticed the problem too and following commits are trying to address that.